src/rule.rs

```diff
@@ -133,12 +153,44 @@ pub struct RuleSubvolume {
 }
 
 impl RuleSubvolume {
-    fn snapshot(&self) -> anyhow::Result<()> {
+    fn snapshot(&self, rule: &Rule) -> anyhow::Result<()> {
         let mut name = RandomName::new();
         for _ in 0..16 {
             let target_path = self.target_dir.join(name.as_str());
+            if !rule.pre_snapshot.is_empty() {
+                let mut cmd = std::process::Command::new("sh");
+                cmd.arg("-c").arg(format!("{} {} {}", rule.pre_snapshot, self.path.display(), target_path.display()));
```

Comment:
We probably also need to document this somewhere. We probably want to also have an option for letting this script run in root/user mode. For now I think you can do a simple check to make sure this file is only writable by root

Reply:
Is the check if the script is only writable by root so that someone without root can't use it for executing things as root ?
That would be an issue though even if you meant the check for something else, I will add that.

Reply:
It's so that someone without root can't modify the script.